diff -Nuarp opencv-3.4.20.orig/cmake/OpenCVDetectCXXCompiler.cmake opencv-3.4.20/cmake/OpenCVDetectCXXCompiler.cmake
--- opencv-3.4.20.orig/cmake/OpenCVDetectCXXCompiler.cmake	2023-06-27 19:29:13.000000000 +0800
+++ opencv-3.4.20/cmake/OpenCVDetectCXXCompiler.cmake	2024-06-10 14:54:29.392187073 +0800
@@ -143,9 +143,9 @@ elseif(MSVC)
   # see Modules/CMakeGenericSystem.cmake
   if("${CMAKE_GENERATOR}" MATCHES "(Win64|IA64)")
     set(OpenCV_ARCH "x64")
-  elseif("${CMAKE_GENERATOR_PLATFORM}" MATCHES "ARM64")
+  elseif("${CMAKE_GENERATOR_PLATFORM}" MATCHES "(ARM64|arm64)")
     set(OpenCV_ARCH "ARM64")
-  elseif("${CMAKE_GENERATOR}" MATCHES "ARM")
+  elseif("${CMAKE_GENERATOR_PLATFORM}" MATCHES "(ARM|arm)")
     set(OpenCV_ARCH "ARM")
   elseif("${CMAKE_SIZEOF_VOID_P}" STREQUAL "8")
     set(OpenCV_ARCH "x64")
diff -Nuarp opencv-3.4.20.orig/cmake/templates/OpenCVConfig.root-WIN32.cmake.in opencv-3.4.20/cmake/templates/OpenCVConfig.root-WIN32.cmake.in
--- opencv-3.4.20.orig/cmake/templates/OpenCVConfig.root-WIN32.cmake.in	2023-06-27 19:29:13.000000000 +0800
+++ opencv-3.4.20/cmake/templates/OpenCVConfig.root-WIN32.cmake.in	2024-06-10 14:54:43.583157025 +0800
@@ -99,9 +99,9 @@ elseif(MSVC)
   # see Modules/CMakeGenericSystem.cmake
   if("${CMAKE_GENERATOR}" MATCHES "(Win64|IA64)")
     set(OpenCV_ARCH "x64")
-  elseif("${CMAKE_GENERATOR_PLATFORM}" MATCHES "ARM64")
+  elseif("${CMAKE_GENERATOR_PLATFORM}" MATCHES "(ARM64|arm64)")
     set(OpenCV_ARCH "ARM64")
-  elseif("${CMAKE_GENERATOR}" MATCHES "ARM")
+  elseif("${CMAKE_GENERATOR_PLATFORM}" MATCHES "(ARM|arm)")
     set(OpenCV_ARCH "ARM")
   elseif("${CMAKE_SIZEOF_VOID_P}" STREQUAL "8")
     set(OpenCV_ARCH "x64")
