diff -Nuarp opencv-4.14.0.orig/modules/core/src/persistence.hpp opencv-4.14.0/modules/core/src/persistence.hpp
--- opencv-4.14.0.orig/modules/core/src/persistence.hpp	2026-09-25 14:12:25.290613543 +0800
+++ opencv-4.14.0/modules/core/src/persistence.hpp	2026-09-25 14:14:27.453061245 +0800
@@ -11,7 +11,7 @@
 #include <string>
 #include <iterator>
 
-#define USE_ZLIB 1
+#define USE_ZLIB 0
 #if USE_ZLIB
 #  ifndef _LFS64_LARGEFILE
 #    define _LFS64_LARGEFILE 0
