diff -Nuarp opencv-4.10.0.orig/cmake/OpenCVGenAndroidMK.cmake opencv-4.10.0/cmake/OpenCVGenAndroidMK.cmake
--- opencv-4.10.0.orig/cmake/OpenCVGenAndroidMK.cmake	2024-06-02 04:41:06.000000000 +0800
+++ opencv-4.10.0/cmake/OpenCVGenAndroidMK.cmake	2024-06-04 21:02:30.323531599 +0800
@@ -76,6 +76,4 @@ if(ANDROID)
 
   configure_file("${OpenCV_SOURCE_DIR}/cmake/templates/OpenCV.mk.in" "${CMAKE_BINARY_DIR}/unix-install/OpenCV.mk" @ONLY)
   configure_file("${OpenCV_SOURCE_DIR}/cmake/templates/OpenCV-abi.mk.in" "${CMAKE_BINARY_DIR}/unix-install/OpenCV-${ANDROID_NDK_ABI_NAME}.mk" @ONLY)
-  install(FILES ${CMAKE_BINARY_DIR}/unix-install/OpenCV.mk DESTINATION ${OPENCV_CONFIG_INSTALL_PATH} COMPONENT dev)
-  install(FILES ${CMAKE_BINARY_DIR}/unix-install/OpenCV-${ANDROID_NDK_ABI_NAME}.mk DESTINATION ${OPENCV_CONFIG_INSTALL_PATH} COMPONENT dev)
 endif(ANDROID)
diff -Nuarp opencv-4.10.0.orig/cmake/OpenCVUtils.cmake opencv-4.10.0/cmake/OpenCVUtils.cmake
--- opencv-4.10.0.orig/cmake/OpenCVUtils.cmake	2024-06-02 04:41:06.000000000 +0800
+++ opencv-4.10.0/cmake/OpenCVUtils.cmake	2024-06-04 21:02:50.020458607 +0800
@@ -976,11 +976,6 @@ macro(ocv_finalize_status)
     endif()
   endif()
 
-  if(UNIX)
-    install(FILES "${OpenCV_SOURCE_DIR}/platforms/scripts/valgrind.supp"
-                  "${OpenCV_SOURCE_DIR}/platforms/scripts/valgrind_3rdparty.supp"
-            DESTINATION "${OPENCV_OTHER_INSTALL_PATH}" COMPONENT "dev")
-  endif()
 endmacro()
 
 
diff -Nuarp opencv-4.10.0.orig/CMakeLists.txt opencv-4.10.0/CMakeLists.txt
--- opencv-4.10.0.orig/CMakeLists.txt	2024-06-02 04:41:06.000000000 +0800
+++ opencv-4.10.0/CMakeLists.txt	2024-06-04 21:03:59.252202053 +0800
@@ -1076,20 +1076,12 @@ ocv_add_modules_compiler_options()
 ocv_register_modules()
 
 # Generate targets for documentation
-add_subdirectory(doc)
 
 # various data that is used by cv libraries and/or demo applications.
-add_subdirectory(data)
 
 # extra applications
-if(BUILD_opencv_apps)
-  add_subdirectory(apps)
-endif()
 
 # examples
-if(BUILD_EXAMPLES OR BUILD_ANDROID_EXAMPLES OR INSTALL_ANDROID_EXAMPLES OR INSTALL_PYTHON_EXAMPLES OR INSTALL_C_EXAMPLES)
-  add_subdirectory(samples)
-endif()
 
 # ----------------------------------------------------------------------------
 # Finalization: generate configuration-based files
@@ -1101,20 +1093,13 @@ ocv_cmake_hook(PRE_FINALIZE)
 include(cmake/OpenCVGenHeaders.cmake)
 
 # Generate opencv.pc for pkg-config command
-if(OPENCV_GENERATE_PKGCONFIG)
-  include(cmake/OpenCVGenPkgconfig.cmake)
-endif()
 
 # Generate OpenCV.mk for ndk-build (Android build tool)
-include(cmake/OpenCVGenAndroidMK.cmake)
 
 # Generate OpenCVConfig.cmake and OpenCVConfig-version.cmake for cmake projects
 include(cmake/OpenCVGenConfig.cmake)
 
 # Generate Info.plist for the iOS/visionOS framework
-if(APPLE_FRAMEWORK)
-  include(cmake/OpenCVGenInfoPlist.cmake)
-endif()
 
 # Generate ABI descriptor
 include(cmake/OpenCVGenABI.cmake)
@@ -1140,12 +1125,6 @@ if(INSTALL_TESTS AND OPENCV_TEST_DATA_PA
   endif()
 endif()
 
-if(NOT OPENCV_README_FILE)
-  if(ANDROID)
-    set(OPENCV_README_FILE ${CMAKE_CURRENT_SOURCE_DIR}/platforms/android/README.android)
-  endif()
-endif()
-
 if(NOT OPENCV_LICENSE_FILE)
   set(OPENCV_LICENSE_FILE ${CMAKE_CURRENT_SOURCE_DIR}/LICENSE)
 endif()
@@ -1155,11 +1134,6 @@ if(ANDROID OR NOT UNIX)
   install(FILES ${OPENCV_LICENSE_FILE}
         PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
         DESTINATION ./ COMPONENT libs)
-  if(OPENCV_README_FILE)
-    install(FILES ${OPENCV_README_FILE}
-            PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
-            DESTINATION ./ COMPONENT libs)
-  endif()
 endif()
 
 if(COMMAND ocv_pylint_finalize)
diff -Nuarp opencv-4.10.0.orig/data/CMakeLists.txt opencv-4.10.0/data/CMakeLists.txt
--- opencv-4.10.0.orig/data/CMakeLists.txt	2024-06-02 04:41:06.000000000 +0800
+++ opencv-4.10.0/data/CMakeLists.txt	2024-06-04 21:04:10.806159236 +0800
@@ -1,9 +1,6 @@
 file(GLOB HAAR_CASCADES haarcascades/*.xml)
 file(GLOB LBP_CASCADES lbpcascades/*.xml)
 
-install(FILES ${HAAR_CASCADES} DESTINATION ${OPENCV_OTHER_INSTALL_PATH}/haarcascades COMPONENT libs)
-install(FILES ${LBP_CASCADES}  DESTINATION ${OPENCV_OTHER_INSTALL_PATH}/lbpcascades  COMPONENT libs)
-
 if(INSTALL_TESTS AND OPENCV_TEST_DATA_PATH)
   install(DIRECTORY "${OPENCV_TEST_DATA_PATH}/" DESTINATION "${OPENCV_TEST_DATA_INSTALL_PATH}" COMPONENT "tests")
 endif()
