diff -Nuarp opencv-5.0.0.orig/CMakeLists.txt opencv-5.0.0/CMakeLists.txt
--- opencv-5.0.0.orig/CMakeLists.txt	2026-07-11 00:18:55.562700631 +0800
+++ opencv-5.0.0/CMakeLists.txt	2026-07-11 00:18:59.533896451 +0800
@@ -77,10 +77,12 @@ if(OPENCV_WORKAROUND_CMAKE_20989)
   set(CMAKE_SYSTEM_PROCESSOR ${CMAKE_SYSTEM_PROCESSOR_BACKUP})
 endif()
 
-include(CheckLanguage)
-check_language(ASM)
-if(CMAKE_ASM_COMPILER)
-  enable_language(ASM)
+if(NOT MSVC)
+  include(CheckLanguage)
+  check_language(ASM)
+  if(CMAKE_ASM_COMPILER)
+    enable_language(ASM)
+  endif()
 endif()
 
 enable_testing()
