diff -Nuarp opencv-4.9.0.orig/modules/core/src/persistence.hpp opencv-4.9.0/modules/core/src/persistence.hpp
--- opencv-4.9.0.orig/modules/core/src/persistence.hpp	2023-12-28 00:46:55.000000000 +0800
+++ opencv-4.9.0/modules/core/src/persistence.hpp	2024-01-17 19:26:39.076216559 +0800
@@ -12,7 +12,7 @@
 #include <string>
 #include <iterator>
 
-#define USE_ZLIB 1
+#define USE_ZLIB 0
 #if USE_ZLIB
 #  ifndef _LFS64_LARGEFILE
 #    define _LFS64_LARGEFILE 0
