diff -Nuarp opencv-4.14.0.orig/modules/core/src/mathfuncs.cpp opencv-4.14.0/modules/core/src/mathfuncs.cpp
--- opencv-4.14.0.orig/modules/core/src/mathfuncs.cpp	2026-09-25 14:12:25.287961392 +0800
+++ opencv-4.14.0/modules/core/src/mathfuncs.cpp	2026-09-25 14:14:47.968589803 +0800
@@ -1972,8 +1972,13 @@ const double* getExpTab64f()
 const float* getExpTab32f()
 {
     static float CV_DECL_ALIGNED(64) expTab_f[EXPTAB_MASK+1];
+#ifndef OPENCV_DISABLE_THREAD_SUPPORT
     static std::atomic<bool> expTab_f_initialized(false);
     if (!expTab_f_initialized.load())
+#else
+    static bool expTab_f_initialized = false;
+    if (!expTab_f_initialized)
+#endif
     {
         for( int j = 0; j <= EXPTAB_MASK; j++ )
             expTab_f[j] = (float)expTab[j];
diff -Nuarp opencv-4.14.0.orig/modules/core/src/parallel.cpp opencv-4.14.0/modules/core/src/parallel.cpp
--- opencv-4.14.0.orig/modules/core/src/parallel.cpp	2026-09-25 14:14:06.743475868 +0800
+++ opencv-4.14.0/modules/core/src/parallel.cpp	2026-09-25 14:14:47.968721844 +0800
@@ -493,6 +493,7 @@ void parallel_for_(const cv::Range& rang
     if (range.empty())
         return;
 
+#ifndef OPENCV_DISABLE_THREAD_SUPPORT
     static std::atomic<bool> flagNestedParallelFor(false);
     bool isNotNestedRegion = !flagNestedParallelFor.load();
     if (isNotNestedRegion)
@@ -505,6 +506,7 @@ void parallel_for_(const cv::Range& rang
         }
     }
     else // nested parallel_for_() calls are not parallelized
+#endif
     {
         CV_UNUSED(nstripes);
         body(range);
