diff -Nuarp opencv-4.8.0.orig/modules/core/src/persistence.hpp opencv-4.8.0/modules/core/src/persistence.hpp
--- opencv-4.8.0.orig/modules/core/src/persistence.hpp	2023-06-28 19:53:33.000000000 +0800
+++ opencv-4.8.0/modules/core/src/persistence.hpp	2023-08-06 11:17:31.634207928 +0800
@@ -12,7 +12,7 @@
 #include <string>
 #include <iterator>
 
-#define USE_ZLIB 1
+#define USE_ZLIB 0
 #if USE_ZLIB
 #  ifndef _LFS64_LARGEFILE
 #    define _LFS64_LARGEFILE 0
