diff -Nuarp opencv-3.4.16/modules/core/src/persistence.hpp opencv-3.4.16.nozlib/modules/core/src/persistence.hpp
--- opencv-3.4.16/modules/core/src/persistence.hpp	2021-10-08 11:31:56.000000000 +0800
+++ opencv-3.4.16.nozlib/modules/core/src/persistence.hpp	2021-11-24 10:58:06.142465885 +0800
@@ -12,7 +12,7 @@
 #include <string>
 #include <iterator>
 
-#define USE_ZLIB 1
+#define USE_ZLIB 0
 #if USE_ZLIB
 #  ifndef _LFS64_LARGEFILE
 #    define _LFS64_LARGEFILE 0
