diff -Nuarp opencv-3.4.20.orig/modules/core/include/opencv2/core/hal/intrin_wasm.hpp opencv-3.4.20/modules/core/include/opencv2/core/hal/intrin_wasm.hpp
--- opencv-3.4.20.orig/modules/core/include/opencv2/core/hal/intrin_wasm.hpp	2023-06-27 19:29:13.000000000 +0800
+++ opencv-3.4.20/modules/core/include/opencv2/core/hal/intrin_wasm.hpp	2023-08-06 19:09:44.971191540 +0800
@@ -8,6 +8,7 @@
 #include <limits>
 #include <cstring>
 #include <algorithm>
+#include <emscripten/version.h>
 #include "opencv2/core/saturate.hpp"
 
 #define CV_SIMD128 1
