diff -Nuarp opencv-4.6.0.orig/modules/core/src/persistence.hpp opencv-4.6.0/modules/core/src/persistence.hpp
--- opencv-4.6.0.orig/modules/core/src/persistence.hpp	2022-06-05 08:32:44.000000000 +0800
+++ opencv-4.6.0/modules/core/src/persistence.hpp	2022-11-16 17:40:26.104045965 +0800
@@ -12,7 +12,7 @@
 #include <string>
 #include <iterator>
 
-#define USE_ZLIB 1
+#define USE_ZLIB 0
 #if USE_ZLIB
 #  ifndef _LFS64_LARGEFILE
 #    define _LFS64_LARGEFILE 0
