diff -Nuarp opencv-3.4.13.orig/modules/core/src/persistence.hpp opencv-3.4.13/modules/core/src/persistence.hpp
--- opencv-3.4.13.orig/modules/core/src/persistence.hpp	2020-12-20 14:15:48.000000000 +0800
+++ opencv-3.4.13/modules/core/src/persistence.hpp	2021-02-24 23:55:08.440237914 +0800
@@ -12,7 +12,7 @@
 #include <string>
 #include <iterator>
 
-#define USE_ZLIB 1
+#define USE_ZLIB 0
 #if USE_ZLIB
 #  ifndef _LFS64_LARGEFILE
 #    define _LFS64_LARGEFILE 0
