diff -Nuarp opencv-4.5.4/modules/core/src/persistence.hpp opencv-4.5.4.nozlib/modules/core/src/persistence.hpp
--- opencv-4.5.4/modules/core/src/persistence.hpp	2021-10-09 08:48:26.000000000 +0800
+++ opencv-4.5.4.nozlib/modules/core/src/persistence.hpp	2021-11-24 11:10:47.321454073 +0800
@@ -12,7 +12,7 @@
 #include <string>
 #include <iterator>
 
-#define USE_ZLIB 1
+#define USE_ZLIB 0
 #if USE_ZLIB
 #  ifndef _LFS64_LARGEFILE
 #    define _LFS64_LARGEFILE 0
