diff -Nuarp opencv-2.4.13.7.orig/modules/core/src/persistence.cpp opencv-2.4.13.7/modules/core/src/persistence.cpp
--- opencv-2.4.13.7.orig/modules/core/src/persistence.cpp	2018-07-02 05:41:56.000000000 +0800
+++ opencv-2.4.13.7/modules/core/src/persistence.cpp	2021-02-25 00:07:23.031002590 +0800
@@ -47,17 +47,7 @@
 #include <iterator>
 #include <wchar.h>
 
-#define USE_ZLIB 1
-
-#ifdef __APPLE__
-#  include "TargetConditionals.h"
-#  if (defined TARGET_OS_IPHONE && TARGET_OS_IPHONE) || (defined TARGET_IPHONE_SIMULATOR && TARGET_IPHONE_SIMULATOR)
-#    undef USE_ZLIB
-#    define USE_ZLIB 0
-     typedef void* gzFile;
-#  endif
-#endif
-
+#define USE_ZLIB 0
 #if USE_ZLIB
 #  ifndef _LFS64_LARGEFILE
 #    define _LFS64_LARGEFILE 0
@@ -66,6 +56,8 @@
 #    define _FILE_OFFSET_BITS 0
 #  endif
 #  include <zlib.h>
+#else
+typedef void* gzFile;
 #endif
 
 /****************************************************************************************\
